addons/bitbucket/models.py

```diff
@@ -309,34 +314,45 @@ def before_page_load(self, node, user):
         if self.user_settings is None:
             return messages
 
-        node_permissions = 'public' if node.is_public else 'private'
-        repo_permissions = 'private' if self.is_private else 'public'
-        if repo_permissions != node_permissions:
+        if self.fetched_repo:
+            node_permissions = 'public' if node.is_public else 'private'
+            repo_permissions = 'private' if self.is_private else 'public'
```

Comment:
You're making a duplicate request here. `fetched_repo` makes a call to `.repo()` and `.is_private` makes the same request (see the confusion in making requests in properties? =)).

You could reuse the response from `.fetch_repo()`. Something like

```python
repo_data = self.fetch_repo()
... if repo_data['is_private']
```